commons-app/apps-android-commons
# Multiple uploads with over haul

## Title (required)

Fixes #604
Fixes #1128

## Description (required)

Continued on top of #1796. 

- upload overhaul
- in-app multiple uploads

## Tests performed (required)

Uploads: 
- Camera upload > done
- Camera nearby upload with successful wikidata edit > done
- Single gallery upload > done
- Multiple gallery upload > done
- Gallery upload with successful wikidata edit > done

In the current ShareActivity:

- Tooltips for title, desc, license fields > done
- Language selector for desc > done
- Prefilling of title and desc when upload initiated via Nearby > done
- License selector > done
- Link to license and Commons policies > done
- Zoom in (and back out) > done
- Open location of image on maps > done

Category suggestions/selection:

- When search text field is empty, display:
    - Recent categories > done
    - Location-based category suggestions (if image has geolocation) > not working
    - Title-based category suggestions > not working
    - Wikidata item category (if image is uploaded via Nearby and if that Wikidata item has a category) > not working

- When user types in search text field:
    - Check for categories shortly after typing stops, repeatedly > done
    - Display exact matches at the top > done
    - Categories selected via both of these suggestion types need to be added to the list of category selections > done

Warnings:

- If image is too dark > done
- If image already exists in Commons (via hash check) > done
- If user tries to submit empty title (do not allow to proceed) > The submit button is disabled for this case
- If user does not select any categories


Image templates:
Ensure the same templates are added (not sure if this will be an issue if backend has not changed, but make sure you check). > done
